app/representers/api/image_representer.rb

```diff
@@ -2,12 +2,30 @@
 
 class Api::ImageRepresenter < Api::BaseRepresenter
 
+  def self_url(represented)
+    if represented.is_a?(StoryImage)
+      api_story_story_image_path(represented.story, represented)
+    elsif represented.is_a?(SeriesImage)
+      api_series_series_image_path(represented.series, represented)
+    elsif represented.is_a?(AccountImage)
+      api_account_account_image_path(represented.account, represented)
+    elsif represented.is_a?(UserImage)
+      api_user_user_image_path(represented.user)
+    else
```

Comment:
Yeah, I started down that road.  But the singular `UserImage` (where everything else is plural) makes that a bit odd.  So I opted for explicit-everything.